tests/queries/0_stateless/02167_format_from_file_extension.sh

```diff
@@ -32,3 +32,30 @@ $CLICKHOUSE_CLIENT -q "insert into table function file('test_02167.messagepack',
 $CLICKHOUSE_CLIENT -q "select * from file('test_02167.messagepack') settings input_format_msgpack_number_of_columns=1"
 $CLICKHOUSE_CLIENT -q "select * from file('test_02167.messagepack', 'MsgPack', 'x UInt64')"
 
+for format in TSV TabSeparated TSVWithNames TSVWithNamesAndTypes CSV Parquet ORC Arrow JSONEachRow JSONCompactEachRow CustomSeparatedWithNamesAndTypes
+do
+    $CLICKHOUSE_CLIENT -q "insert into table function file('test_like_02167.$format', 'auto', 'x UInt64') select * from numbers(2)"
+    $CLICKHOUSE_CLIENT -q "select * from file('test_like_02167.$format')"
+    $CLICKHOUSE_CLIENT -q "select * from file('test_like_02167.$format', '$format')"
+done
+
+$CLICKHOUSE_CLIENT -q "insert into table function file('test_like_02167.bin', 'auto', 'x UInt64') select * from numbers(2)"
+$CLICKHOUSE_CLIENT -q "select * from file(['test_02167.bin', 'test_like_02167.bin'], 'auto', 'x UInt64')"
+$CLICKHOUSE_CLIENT -q "select * from file(['test_02167.bin', 'test_like_02167.bin'], 'RowBinary', 'x UInt64')"
+
+$CLICKHOUSE_CLIENT -q "insert into table function file('test_like_02167.ndjson', 'auto', 'x UInt64') select * from numbers(2)"
+$CLICKHOUSE_CLIENT -q "select * from file(['test_02167.ndjson', 'test_like_02167.ndjson'])"
+$CLICKHOUSE_CLIENT -q "select * from file(['test_02167.ndjson', 'test_like_02167.ndjson'], 'JSONEachRow', 'x UInt64')"
+
+$CLICKHOUSE_CLIENT -q "insert into table function file('test_like_02167.messagepack', 'auto', 'x UInt64') select * from numbers(2)"
+$CLICKHOUSE_CLIENT -q "select * from file(['test_02167.messagepack','test_like_02167.messagepack']) settings input_format_msgpack_number_of_columns=1"
+$CLICKHOUSE_CLIENT -q "select * from file(['test_02167.messagepack','test_like_02167.messagepack'], 'MsgPack', 'x UInt64')"
+
+$CLICKHOUSE_CLIENT -q "select * from file(['test_02167.bin'], 'auto', 'x UInt64')"
+$CLICKHOUSE_CLIENT -q "select * from file(['test_02167.bin'], 'RowBinary', 'x UInt64')"
+
+$CLICKHOUSE_CLIENT -q "select * from file(['test_02167.ndjson'])"
+$CLICKHOUSE_CLIENT -q "select * from file(['test_02167.ndjson'], 'JSONEachRow', 'x UInt64')"
+
+$CLICKHOUSE_CLIENT -q "select * from file(['test_02167.messagepack']) settings input_format_msgpack_number_of_columns=1"
+$CLICKHOUSE_CLIENT -q "select * from file(['test_02167.messagepack'], 'MsgPack', 'x UInt64')"
```

Comment:
Tests for cases where empty Array (`[]`) and Array of unsupported type (`[1,2,3]`) should be added to check that we handle them correctly.

Also, to improve the readability of the reference file we can try to add the format name into SELECTs as a parameter:
```
$CLICKHOUSE_CLIENT --param_format="JSONEachRow" -q "select {format:String}, * from file(['test_02167.ndjson'])"
$CLICKHOUSE_CLIENT --param_format="JSONEachRow" -q "select {format:String}, * from file(['test_02167.ndjson'], {format:String}, 'x UInt64')"
```